aten/src/ATen/CachedTensorUtils.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+namespace at {
+namespace caching {
+
+// Some systems (just cudagraphs currently) will persist a static tensor output
+// whose TensorImpl does not change across iterations. For these tensors caching
+// dtype conversions is invalid. Additionally, there will be an extra reference
+// count to these cached tensors that would prevent buffer inplacing and other
+// checks on tensor uniqueness. If we are not using these systems the enabled
+// flag will be false and we will avoid the hash lookup.
+
+TORCH_API bool is_cached_tensor(const at::Tensor& t);
+TORCH_API void add_cached_tensor(const at::Tensor& t);
```

Comment:
What would happen if you add_cached_tensor a tensor twice?

Reply:
We're hashing on the TensorImpl, would be the same as inserting once

Reply:
But then if you remove it, I clear all of them? That seems troublesome? Wouldn't it make more sense for any tensor to only be addable once?

Reply:
It's a set and it has the same semantics as a set. I added a test to verify that it works as intended.